### DISCUSSION TITLE
Allow creation of table names using letters in capital case anywhere in the name

### DISCUSSION THREAD

**Discussion by aschoijett** [directus/directus#3481] [Archive]
Hi,
When I create a new collection, Directus creates a _table_ in my database all in small letters.  I think that having Directus name a _table_ all in small letters is the intended feature, but it is one that goes against our naming conventions.   
Directus' convention may work well if one never has to access the database using any tool other than Directus, but that's not realistic. 

I imagine we're not the only ones who use Capitals to separate parts of table names.   Actually, we not only do that but also prepend a small letter to the table name to indicate the table type.  For example, a master table that keeps all the versions of a product would be named mProductVersion (and not mproductversion) while a transactional _table_ would be called something like xOrderItem (and not xorderitem).   

I realize that Directus can alias the name of the table and show it any way I want in the administrative application, but the database tables are accessed not only through Directus and so I'd like to have them created by directs using the name as I type it.   

Although I think that one can work around that "feature" by renaming the table that Directus creates in a manner that follows our convention, and then going over some adjustment process, doing that seems silly if all that would be needed is to allow the table to be named the way the administrator wants it.   

**Would it be possible to change Directus' Settings > Data Model module to allow us to name the _table_  that would be created in the database anyway we want?**   

Many thanks!

Alex

**Comment by rijkvanzanten** [comment]
The reason why Directus defaults to all lower case is that capitalization in table/column names varies wildly between vendors and operating systems (f.e. MySQL under Windows is case insensitive at all items, where under Unix it's case sensitive. Likewise, IIRC the SQL standard itself specs all upper case names, etc). Long story short, we'll have to make sure it actually works cross platform before we can commit to opening it up in Directus

**Comment by aschoijett** [comment]
Thank you Rijk,

I think my request could still be honoured.   If the target us Unix/Linux,
then if I name a collection mProductVersion, that is how my table would be
named.   The same would be true for Windows.

From what I read in your note, the biggest issue would be that Unix would
allow me to create a 2nd table and call it mPRODUCTversion (a really bad
idea), while Windows would not.   I think the chances of someone needing to
do that for any serious reason are really, really low.

Your thoughts on how I should proceed are really appreciated.  One more
thing, we already have a system built using a database that follows this
convention and we would like to plug Directus to it to manage many of the
existing master tables and a few new ones that we'll be needing soon.











*Alex Schoijett*
Libra Enterprise Consulting <http://www.libragroup.ca>
647-924-7253

**Reply by rijkvanzanten** [reply]
It becomes a question of 80/20 here. Like what is the percentage of users that need—like in your case—support for creating new capitalized tables versus the amount of people potentially running into an issue with mixed cased names..

FWIW

> One more
thing, we already have a system built using a database that follows this
convention and we would like to plug Directus to it to manage many of the
existing master tables and a few new ones that we'll be needing soon

Directus _should_ still work with mixed-cased names, it just doesn't let you type them in in the admin itself. For right now, you should be able to create the tables directly in the DB, and use them in Directus like normal 🙂 In the meantime, I'll do a little more research into the potential downsides of opening this up a bit 👍

**Comment by aschoijett** [comment]
Much appreciated!

*Alex Schoijett*
Libra Enterprise Consulting <http://www.libragroup.ca>
647-924-7253
